traits/tests/test_extended_trait_change.py

```diff
@@ -602,8 +607,16 @@ def test_instance_dict_value(self):
             exp_new={0: 0, 1: 1, 2: 2, 3: 3},
             dst_new={0: 0, 1: 1, 2: 2, 3: 3},
         )
-        inst.ref.value[3] = 3
-        self.assertEqual(inst.calls, {0: 1, 1: 0, 2: 0, 3: 0, 4: 0})
+        with self.assertRaises(AssertionError):
+            # Expected failure, see enthought/traits#537
+            # InstanceValueListener.arg_check1 receives a TraitDictEvent
+            # as `new` instead of the expected `{0: 0, 1: 1, 2: 2, 3: 3}`
+            inst.ref.value[3] = 3
+
+        with self.assertRaises(AssertionError):
+            # Expected failure, see enthought/traits#537
+            self.assertEqual(inst.calls, {0: 1, 1: 1, 2: 1, 3: 1, 4: 1})
```

Comment:
I changed the expected value here. If the bug was fixed, we expect all change handlers to be called, right?

Reply:
Actually, reading #537 again, I am no longer sure what the expected behaviour is. I am not going to make any claim here what the expected behaviour should be.